components/tcf/services/src/index.js

```diff
@@ -0,0 +1,22 @@
+import {useEffect, useState} from 'react'
+import PropTypes from 'prop-types'
+import {ServiceInitializer} from './infrastructure/bootstrap/ServiceInitializer'
+
+export default function TcfServices({children}) {
+  const [service, setService] = useState(false)
+
+  useEffect(function() {
+    Promise.resolve()
+      .then(() => ServiceInitializer.init())
+      .then(service => {
+        setService(service)
+      })
+  }, [])
+
+  return service && children(service)
+}
+
+TcfServices.displayName = 'TcfServices'
+TcfServices.propTypes = {
+  children: PropTypes.func.isRequired
```

Comment:
why a function?

Reply:
Check index.js file from ui [component](https://github.com/SUI-Components/adevinta-spain-components/blob/750e7504777da32c9d8d384755f738e458654ed5/components/tcf/ui/src/index.js ) and you will see that children is a function, from lines +8 to +17. To be able to pass to their children service use cases

Reply:
Weird, but ok